src/modules/directory.rs

```diff
@@ -99,7 +99,15 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         String::from("")
     };
 
-    let path_vec = match &repo.and_then(|r| r.workdir.as_ref()) {
+    const BEFORE_ROOT_PATH: usize = 0;
```

Comment:
After looking at the code, I think I'd prefer it if this got factored into a struct with appropriate `impl`s on it. While this works, it gets very hard to read (e.g. I still can't figure out what the conditional on lines 142-146 is supposed to be doing).

Reply:
I forgot the review doesn't apply to blocks: when I say "this", I'm referring to the usage of `const` indices + a vec to store things in to form a simple dictionary. I would prefer if we could just do something like `components.before_root_path`.